src/components/ContentNode.vue

```diff
@@ -107,11 +107,29 @@ function renderNode(createElement, references) {
     renderNode(createElement, references),
   );
 
-  const renderListItems = items => items.map(item => (
-    createElement('li', {}, (
-      renderChildren(item.content)
-    ))
-  ));
+  const renderListItems = items => items.map((item) => {
```

Comment:
This change works great, although it adds a bit of complexity to an already very large render function.

Not a blocker, but would you mind if I possibly take a pass at refactoring this to extract an explicit task-list component that we could also re-use for other things in the future if needed?

Reply:
I worked on this a bit if you're curious (99.9% done) and can open a PR against your fork if you'd like.

[diff against your branch](https://github.com/QuietMisdreavus/swift-docc-render/compare/task-lists...mportiz08:swift-docc-render:task-list-component)
[diff against main](https://github.com/apple/swift-docc-render/compare/main...mportiz08:swift-docc-render:task-list-component)

Reply:
My one reservation about the component as you've written it (if i understand it correctly) is that it's possible to mix "task list items" with plain "list items", e.g. the task list in this PR description. What happens in your patch if someone writes a "mixed list" like that?

Reply:
Ah, I didn't realize it was supposed to be possible to mix/match like that, but I think I realize now why there wasn't a separate list type in the JSON to represent this. I can make one small tweak to not show the checkbox for regular items in that scenario.

I think we would generally want to match the behavior of the GitHub rendering where that item just shows up without the checkbox in that scenario, right?

Example:
- [ ] a
- [x] b
- c
- [ ] d

Reply:
I updated it to work that way with https://github.com/mportiz08/swift-docc-render/commit/c96a2419d6ef1664e990af59a21844c36a21a43d

Reply:
I think that works, and likely matches what people expect. You can push your branch directly into mine if you'd like.